csob/resources/payment/init.py

```diff
@@ -0,0 +1,100 @@
+import json
+from typing import Optional, List
+
+from csob.resources.payment import PaymentCSOBResource
+
+
+class PaymentInit(PaymentCSOBResource):
+    url = 'payment/init'
+    request_signature = (
+        'merchantId', 'orderNo', 'dttm', 'payOperation', 'payMethod', 'totalAmount', 'currency', 'closePayment',
+        'returnUrl', 'returnMethod', 'cart', 'description', 'merchantData', 'customerId', 'language', 'ttlSec',
+        'logoVersion', 'colorSchemeVersion')
+    optional_request_signature = ('merchantData', 'customerId', 'ttlSec', 'logoVersion', 'colorSchemeVersion')
+
+    def post(self, key, order_number: str, pay_operation: str, pay_method: str, total_amount: int, currency: str,
+             close_payment: bool, return_url: str, return_method: str, description: str, language: str,
+             merchant_data: Optional[str], customer_id: Optional[str], cart: Optional[List[dict]] = None,
+             ttl_sec: Optional[int] = None, logo_version: Optional[int] = None,
+             color_scheme_version: Optional[int] = None):
+        local_json = self.get_base_json()
+
+        if len(order_number) > 10:
+            raise ValueError('orderNo is too long.')
+        local_json['orderNo'] = order_number
+
+        if pay_operation not in ['payment', 'oneclickPayment']:
+            raise ValueError('payOperation invalid value')
+        local_json['payOperation'] = pay_operation
+
+        if pay_method != 'card':
+            raise ValueError('payMethod invalid value')
+        local_json['payMethod'] = pay_method
+
+        if type(total_amount) != int:
```

Comment:
nemáš to náhodou výše otypované v hlavičce metody?

Reply:
to python typování není pro samotný python nijak závazné a ten to v podstatě bere jen jako komentář jediné výhody jsou, že je chytřější našeptávač v PyCharmu, člověku by to mělo sloužit víc jako dokumentace toho co se tam dává a když se nad tím pustí ten nástroj `mypy` tak by to mělo odhalit chyby v tom

Reply:
já vím, ale tak nějak mi tohle přijde zbytečně moc okontrolované

Reply:
ok nech to tam, když jsi se s tím už psal